crates/primitives/src/env.rs

```diff
@@ -12,54 +12,165 @@ pub struct Env {
+    /// The value sent to `transact_to`.
     pub value: U256,
+    /// The data of the transaction.
     #[cfg_attr(feature = "serde", serde(with = "crate::utilities::serde_hex_bytes"))]
     pub data: Bytes,
-    pub chain_id: Option<u64>,
+    /// The nonce of the transaction. If set to `None`, no checks are performed.
     pub nonce: Option<u64>,
+
+    /// The chain ID of the transaction. If set to `None`, no checks are performed.
+    ///
+    /// Incorporated as part of the Spurious Dragon upgrade via [EIP-155].
+    ///
+    /// [EIP-155]: https://eips.ethereum.org/EIPS/eip-155
+    pub chain_id: Option<u64>,
+
+    /// A list of addresses and storage keys that the transaction plans to access.
+    ///
+    /// Added in [EIP-2930].
+    ///
+    /// [EIP-2930]: https://eips.ethereum.org/EIPS/eip-2930
     pub access_list: Vec<(B160, Vec<U256>)>,
+
+    /// The priority fee per gas.
+    ///
+    /// Incorporated as part of the London upgrade via [EIP-1559].
+    ///
+    /// [EIP-1559]: https://eips.ethereum.org/EIPS/eip-1559
+    pub gas_priority_fee: Option<U256>,
+
+    /// The list of blob versioned hashes.
+    ///
+    /// Incorporated as part of the Cancun upgrade via [EIP-4844].
+    ///
+    /// [EIP-4844]: https://eips.ethereum.org/EIPS/eip-4844
+    pub blob_hashes: Vec<B256>,
+    /// The max fee per blob gas.
+    ///
+    /// Incorporated as part of the Cancun upgrade via [EIP-4844].
+    ///
+    /// [EIP-4844]: https://eips.ethereum.org/EIPS/eip-4844
+    pub max_fee_per_blob_gas: Option<U256>,
 }
 
+impl TxEnv {
+    /// See [EIP-4844] and [`Env::calc_data_fee`].
+    ///
+    /// [EIP-4844]: https://eips.ethereum.org/EIPS/eip-4844
+    #[inline]
+    pub fn get_total_blob_gas(&self) -> u64 {
+        GAS_PER_BLOB * self.blob_hashes.len() as u64
+    }
+}
+
+/// Transaction destination.
 #[derive(Clone, Debug)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 pub enum TransactTo {
+    /// Simple call to an address.
     Call(B160),
+    /// Contract creation.
     Create(CreateScheme),
 }
 
 impl TransactTo {
+    /// Calls the given address.
+    #[inline]
+    pub fn call(address: B160) -> Self {
+        Self::Call(address)
+    }
+
+    /// Creates a contract.
+    #[inline]
     pub fn create() -> Self {
         Self::Create(CreateScheme::Create)
     }
+
+    /// Creates a contract with the given salt using `CREATE2`.
+    #[inline]
+    pub fn create2(salt: U256) -> Self {
+        Self::Create(CreateScheme::Create2 { salt })
+    }
+
+    /// Returns `true` if the transaction is `Call`.
+    #[inline]
+    pub fn is_call(&self) -> bool {
+        matches!(self, Self::Call(_))
+    }
+
+    /// Returns `true` if the transaction is `Create` or `Create2`.
+    #[inline]
     pub fn is_create(&self) -> bool {
-        matches!(self, Self::Create(_))
+        matches!(
+            self,
+            Self::Create(CreateScheme::Create | CreateScheme::Create2 { .. })
+        )
```

Comment:
Why this change?

Reply:
I prefer exhaustive patterns so when something changes you are reminded to also update everywhere it's used

Reply:
It seems not needed, can you revert this?